tedana/decomposition/gift_pca.py

```diff
@@ -0,0 +1,599 @@
+
+
+def _sumN(dat):
+    """
+    Sum of all the elements of the dat matrix.
+
+    Parameters
+    ----------
+    dat :
+
+    Returns
+    -------
+    u :
+    """
+    return np.sum(dat[:])
+
+
+def _checkOrder(n_in):
+    """
+    Checks the order passed to the window functions.
+
+    Parameters
+    ----------
+    n_in :
+
+    Returns
+    -------
+    n_out :
+    w :
+    trivalwin :
+    """
+
+    w = []
+    trivalwin = 0
+
+    # Special case of negative orders:
+    if n_in < 0:
+        raise ValueError('Order cannot be less than zero.')
+
+    # Check if order is already an integer or empty
+    # If not, round to nearest integer.
+    if not n_in or n_in == np.floor(n_in):
+        n_out = n_in
+    else:
+        n_out = np.round(n_in)
+        LGR.warning('Rounding order to nearest integer.')
+
+    # Special cases:
+    if not n_out or n_out == 0:
+        w = np.zeros((0, 1))  # Empty matrix: 0-by-1
+        trivalwin = 1
+    elif n_out == 1:
+        w = 1
+        trivalwin = 1
+
+    return n_out, w, trivalwin
+
+
+def _parzen_win(n):
+    """
+    Returns the N-point Parzen (de la Valle-Poussin) window in
+    a column vector.
+
+    Parameters
+    ----------
+    n :
+
+    Returns
+    -------
+    w :
+    """
+
+    # Check for valid window length (i.e., n < 0)
+    n, w, trivialwin = _checkOrder(n)
+    if trivialwin:
+        return w
+
+    # Index vectors
+    k = np.arange(-(n - 1) / 2, ((n - 1) / 2) + 1)
+    k1 = k[k < -(n - 1) / 4]
+    k2 = k[abs(k) <= (n - 1) / 4]
+
+    # Equation 37 of [1]: window defined in three sections
+    w1 = 2 * (1 - abs(k1) / (n / 2))**3
+    w2 = 1 - 6 * (abs(k2) / (n / 2))**2 + 6 * (abs(k2) / (n / 2))**3
+    w = np.hstack((w1, w2, w1[::-1])).T
+
+    return w
+
+
+def _entrate_sp(x, sm_window):
+    """
+    Calculate the entropy rate of a stationary Gaussian random process using
+    spectrum estimation with smoothing window.
+
+    Parameters
+    ----------
+    x :
+    sm_window :
+
+    Returns
+    -------
+    out :
+    """
+
+    n = x.shape
+
+    # Normalize x_sb to be unit variance
+    x_std = np.std(np.reshape(x, (np.prod(n), 1)))
+    if x_std < 1e-10:
+        x_std = 1e-10
+    x = x / x_std
+
+    if (sm_window == 1):
+
+        M = [int(i) for i in np.ceil(np.array(n) / 10)]
+
+        if (x.ndim >= 3):
+            parzen_w_3 = np.zeros((2 * n[2] - 1, ))
+            parzen_w_3[(n[2] - M[2] - 1):(n[2] +
+                                          M[2])] = _parzen_win(2 * M[2] + 1)
+
+        if (x.ndim >= 2):
+            parzen_w_2 = np.zeros((2 * n[1] - 1, ))
+            parzen_w_2[(n[1] - M[1] - 1):(n[1] +
+                                          M[1])] = _parzen_win(2 * M[1] + 1)
+
+        if (x.ndim >= 1):
+            parzen_w_1 = np.zeros((2 * n[0] - 1, ))
+            parzen_w_1[(n[0] - M[0] - 1):(n[0] +
+                                          M[0])] = _parzen_win(2 * M[0] + 1)
+
+    if x.ndim == 2 and min(n) == 1:  # 1D
```

Comment:
I would put parentheses, here :smile_cat: 

```suggestion
    if (x.ndim == 2) and (min(n) == 1):  # 1D
```